File: Codechef/Unpleasant Ones.java
import java.util.*;
 class MyClass {
    public static void main(String args[]) {
     Scanner sc=new Scanner(System.in);
     int t=sc.nextInt();
     for(int y=1;y<=t;y++)
     {
         int n=sc.nextInt();
         ArrayList <Long> arr=new ArrayList<Long>();
         ArrayList <Long> arr1=new ArrayList<Long>();
         
         long a[]=new long[n];
         for(int x=0;x<n;x++)
         {
             a[x]=sc.nextLong();
             if(a[x]%2==0)
             {
                 arr.add(a[x]);
             }
             else{
                 arr1.add(a[x]);
             }
         }
         System.out.println(arr.toString().replace("[", "").replace("]", "").replace(",","")+" "+ arr1.toString().replace("[","").replace("]","").replace(",",""));
     }
    }
}

// Input
// 2   
// 3  
// 3 6 5  
// 2   
// 7 6
// Output
// 5 6 3  
// 6 7 

File: Codechef/C++/unpleasant_ones.cpp
#include <iostream>
using namespace std;

int main() {
	int t;
	cin>>t;
	while(t--)
	{
	    int n;
	    cin>>n;
	    int a[n],b[n],c[n],j=0,p=0;
	    for(int i=0;i<n;i++)
	    {
	        cin>>a[i];
	        if(a[i]%2)
	        b[j++]=a[i];
	        else
	        c[p++]=a[i];
	    }
	    p--;
	    j--;
	    for(int i=0;i<n;i++)
	    {
	        if(p>=0)
	        cout<<c[p--]<<" ";
	        else
	        cout<<b[j--]<<" ";
	    }
	    cout<<endl;
	}
	return 0;
}
/*
Input:
2   
3  
3 6 5  
2   
7 6

Output:
5 6 3  
6 7 
*/